In [38]:
import pandas as pd
import numpy as np

Descargar los datos del Titanic desde kaggle.com/c/titanic -
 Unir los tres CSVs (gender_submission, test y train) - 
Responder las siguientes preguntas:
1. ¿Cuántas personas iban en el 
2. ¿Cuántos hombres y mujeres sobrevivieron? 
3. ¿Cuál fue el top 10 de edad que más sobrevieron y el top 10 de edad que no lo lograron? 
4. ¿Cuántos cargos o títulos iban en el barco? Ejemplo: Capitanes, Mrs. Miss, etc. (Acá usarás expresiones regulares) 
5. ¿Cuánto es la sumatoria del valor de los tickets en USD (yep en USD)?

In [2]:
test = pd.read_csv('test.csv')
train = pd.read_csv('train.csv')
gender_submission = pd.read_csv('gender_submission.csv')

In [3]:
test['archive'] = 'test'

In [4]:
gender_submission['archive'] = 'gender_submission'

In [5]:
train['archive'] = 'train'

visualizando los DataSets

In [6]:
test.head(5)

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked archive  
0  34.5      0      0   330911   7.8292   NaN        Q    test  
1  47.0      1      0   363272   7.0000   NaN        S    test  
2  62.0      0      0   240276   9.6875   NaN        Q    test  
3  27.0      0      0   315154   8.6625   NaN        S    test  
4  22.0      1      1  3101298  12.2875   NaN        S    test

In [8]:
train.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked archive  
0      0         A/5 21171   7.2500   NaN        S   train  
1      0          PC 17599  71.2833   C85        C   train  
2      0  STON/O2. 3101282   7.9250   NaN        S   train  
3      0            113803  53.1000  C123        S   train  
4      0            373450   8.0500   NaN        S   train

In [9]:
gender_submission.head(5)

PassengerId  Survived            archive
0          892         0  gender_submission
1          893         1  gender_submission
2          894         0  gender_submission
3          895         0  gender_submission
4          896         1  gender_submission

Uniendo los datasets

In [34]:
titanic = pd.concat([train, gender_submission,test], join='outer')

limpiando la columna PassengerId para que no haya id duplicados.

In [40]:
titanic = titanic.drop_duplicates(subset=['PassengerId'])

¿Cuántas personas iban en el titanic?

In [47]:
total_people = titanic['PassengerId'].value_counts().sum()

In [48]:
print(f'el total de personas en el titanic es : {total_people}')

el total de personas en el titanic es : 1309


¿Cuántos hombres y mujeres sobrevivieron?

In [64]:
titanic['total'] = 1

In [72]:
sobrev_generos = titanic.groupby(by=['Sex','Survived'])[['total']].sum()

La columna Survived contiene dos valores. Donde [0] es Fallecido y [1] es Sobrevivio

In [73]:
sobrev_generos

total
Sex    Survived       
female 0.0          81
       1.0         233
male   0.0         468
       1.0         109

Sobrevieron 233 Mujeres y 109 Hombres

Cuál fue el top 10 de edad que más sobrevieron y el top 10 de edad que no lo lograron?

In [96]:
top_age_survived = titanic[titanic.Survived==1]

In [100]:
top_age_survived = top_age_survived[top_age_survived['Age'].duplicated()]

In [124]:
total_10_survides = top_age_survived['Age'].value_counts().head(10)

In [125]:
print(f'El TOP de edades en sobrevivir es: \n{total_10_survides} ')

El TOP de edades en sobrevivir es: 
24.0    14
36.0    10
35.0    10
27.0    10
22.0    10
30.0     9
18.0     8
19.0     8
32.0     8
31.0     7
Name: Age, dtype: int64 


In [110]:
top_age_not_survived = titanic[titanic.Survived==0]

In [118]:
top_age_not_survived = top_age_not_survived[top_age_not_survived['Age'].duplicated()]

In [121]:
total_10_not_survides = top_age_not_survived['Age'].value_counts().head(10)

In [122]:
print(f'El TOP de edades en no sobrevivir es: \n{total_10_not_survides}')

El TOP de edades en no sobrevivir es: 
21.0    16
28.0    15
25.0    14
18.0    14
22.0    13
19.0    13
24.0    12
30.0    12
20.0     9
26.0     9
Name: Age, dtype: int64


¿Cuántos cargos o títulos iban en el barco? Ejemplo: Capitanes, Mrs. Miss, etc. (Acá usarás expresiones regulares)

In [255]:
title_Pass = titanic

In [257]:
title_Pass = title_Pass.dropna(subset=['Name'])

In [260]:
title_Pass['Title'] = title_Pass.Name.apply(lambda name: name.split(',')[1].split('.')[0].strip())

C:\Users\Juanca\anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [268]:
count_title = title_Pass['Title'].value_counts()

In [270]:
print(f'el total de titulos en el titanic son: \n{count_title}')

el total de titulos en el titanic son: 
Mr              517
Miss            182
Mrs             125
Master           40
Dr                7
Rev               6
Col               2
Mlle              2
Major             2
Capt              1
Ms                1
Jonkheer          1
the Countess      1
Sir               1
Lady              1
Don               1
Mme               1
Name: Title, dtype: int64


¿Cuánto es la sumatoria del valor de los tickets en USD (yep en USD)?

In [271]:
total = titanic['Fare'].sum()

In [273]:
print(f'La sumatoria de los valores de tickets es : ${total}')

La sumatoria de los valores de tickets es : $28693.9493
